### PR TITLE
(fix) DetailsList group header aria-posinset and aria-setsize attributes

### LINKED ISSUE #15355: List components: Missing screen reader index count
There are a couple of issues regarding the index count of items in `List` components:
- `aria-posinset` and `aria-setsize` attributes are being flagged as not allowed in group headers given that they have the `row` role to account for correct `grid` semantics. We should remove them and use `aria-rowindex` and `aria-rowcount` instead.
- Index count is missing for items within a group so the total number of them are not read when navigating to the group (see #11817) nor are their positions in the set read when navigating through them (see #11819). This can be fixed by passing the correct `row` role to these items and setting both `aria-rowindex` and `aria-rowcount`.
- We've also been talking about having the items in the list match the visual display rather than including hidden/collapsed rows and having the same index across both headers and items. This is a tentative solution that is pending further discussion before committing to it. This can be evaluated by the following example where `Group Header 2` is collapsed:

```
Group Header 1      -   rowindex=1   rowcount=6
    List item 1.1   -   rowindex=2   rowcount=6
    List item 1.2   -   rowindex=3   rowcount=6
Group Header 2      -   rowindex=4   rowcount=6
Group Header 3      -   rowindex=5   rowcount=6
    List item 3.1   -   rowindex=6   rowcount=6
```

And when we expand it then the attributes change to reflect it:

```
Group Header 1      -   rowindex=1   rowcount=8
    List item 1.1   -   rowindex=2   rowcount=8
    List item 1.2   -   rowindex=3   rowcount=8
Group Header 2      -   rowindex=4   rowcount=8
    List item 2.1   -   rowindex=5   rowcount=8
    List item 2.2   -   rowindex=6   rowcount=8
Group Header 3      -   rowindex=7   rowcount=8
    List item 3.1   -   rowindex=8   rowcount=8
```

### PR DESCRIPTION
**Pull request by smhigley**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15355
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Updates the Grouped Detailslist to use grid row semantics for index and count on the group header rows. Does not fix the actual index values on nested grid rows.

#### Focus areas to test

- Grouped Detailslist's group headers should have `aria-rowindex` and `aria-rowcount`, both of which should include the header row when calculating index/count
- GroupedList should use `aria-posinset` and `aria-setsize` for group headers

### CODE REVIEW

**Comment by smhigley** [comment]
Screen reader support testing results for the changes in this PR: https://codepen.io/smhigley/full/WNoGBoZ

File: packages/react/src/components/GroupedList/GroupHeader.base.tsx [hunk 1]
```diff
@@ -107,6 +109,9 @@ export class GroupHeaderBase extends React.Component<IGroupHeaderProps, IGroupHe
     if (!group) {
       return null;
     }
+
+    const groupLabel = `${group.name} (${group.count}${group.hasMoreData ? '+' : ''})`;
```
Comment (cschlechty) [inline packages/react/src/components/GroupedList/GroupHeader.base.tsx:113]: not sure how well this will localize

**Review by cschlechty — APPROVED** [review]

File: packages/react/src/components/GroupedList/GroupHeader.base.tsx (hunk 1 above)
Comment (smhigley) [reply packages/react/src/components/GroupedList/GroupHeader.base.tsx:113]: Good point! I copied it more or less from `_onRenderTitle` (minus the wrapping nodes), so I suppose it'll at least be consistent? 😅. Would it be better to just have the `group.count` after the name, and leave off the "+" sign?

Comment (ThomasMichon) [reply packages/react/src/components/GroupedList/GroupHeader.base.tsx:113]: You should take in a function as props (passed to the `DetailsList` or whatever by the host) which accepts these fields and produces the final string:
```ts
const groupLabel = props.getGroupAriaLabel?({
  name: group.name,
  count: group.count,
  hasMoreData: group.hasMoreData
 });
If it's not present, I'm sorry but we should just use `ariaLabel` directly.

**Review by ThomasMichon — CHANGES_REQUESTED** [review]

File: packages/react/src/components/GroupedList/GroupHeader.base.tsx (hunk 1 above)
Comment (smhigley) [reply packages/react/src/components/GroupedList/GroupHeader.base.tsx]: @cschlechty / @ThomasMichon I updated the PR to use `aria-labelledby` pointing at the group title cell. It's still author-overridable through `group.ariaLabel`, and I think it sets a better default by avoiding separate visible vs. screen reader text.

Let me know if that addresses your concerns! :)

**Review by ThomasMichon — APPROVED** [review]

**Pull request merged**